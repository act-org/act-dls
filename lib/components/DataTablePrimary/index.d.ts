/**
 * @prettier
 */
import * as React from 'react';
import { Props as EmptyStatePrimaryProps } from '~/components/EmptyStatePrimary';
import { SortObject } from '~/types';
interface Column<T> {
    renderValue: (item: T) => any;
    label?: string;
    sortBy?: string;
    style?: React.CSSProperties;
}
export interface Props<T> {
    columns: Column<T>[];
    currentSortObject: SortObject;
    emptyStateProps: EmptyStatePrimaryProps;
    items: T[];
    onChangeSort: (sortObject: SortObject) => void;
    RowWrapper?: (item: T, children: React.ReactElement<any>) => React.ReactElement<any>;
}
declare const DataTablePrimary: {
    <T>({ columns, currentSortObject, emptyStateProps, items, onChangeSort, RowWrapper, }: Props<T>): React.ReactElement<any>;
    defaultProps: {
        RowWrapper: undefined;
    };
};
export default DataTablePrimary;
//# sourceMappingURL=index.d.ts.map