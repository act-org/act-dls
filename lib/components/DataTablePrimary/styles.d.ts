/**
 * @prettier
 */
declare const _default: (props?: any) => Record<"emptyStateContainer", string>;
export default _default;
//# sourceMappingURL=styles.d.ts.map