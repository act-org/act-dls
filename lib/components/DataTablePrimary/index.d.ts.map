{"version":3,"file":"index.d.ts","sourceRoot":"","sources":["../../../src/components/DataTablePrimary/index.tsx"],"names":[],"mappings":"AAAA;;GAEG;AAEH,OAAO,KAAK,KAAK,MAAM,OAAO,CAAC;AAG/B,OAA0B,EACxB,KAAK,IAAI,sBAAsB,EAChC,MAAM,gCAAgC,CAAC;AACxC,OAAO,EAAE,UAAU,EAAE,MAAM,SAAS,CAAC;AAOrC,UAAU,MAAM,CAAC,CAAC;IAChB,WAAW,EAAE,CAAC,IAAI,EAAE,CAAC,KAAK,GAAG,CAAC;IAC9B,KAAK,CAAC,EAAE,MAAM,CAAC;IACf,MAAM,CAAC,EAAE,MAAM,CAAC;IAChB,KAAK,CAAC,EAAE,KAAK,CAAC,aAAa,CAAC;CAC7B;AAED,MAAM,WAAW,KAAK,CAAC,CAAC;IACtB,OAAO,EAAE,MAAM,CAAC,CAAC,CAAC,EAAE,CAAC;IACrB,iBAAiB,EAAE,UAAU,CAAC;IAC9B,eAAe,EAAE,sBAAsB,CAAC;IACxC,KAAK,EAAE,CAAC,EAAE,CAAC;IACX,YAAY,EAAE,CAAC,UAAU,EAAE,UAAU,KAAK,IAAI,CAAC;IAC/C,UAAU,CAAC,EAAE,CACX,IAAI,EAAE,CAAC,EACP,QAAQ,EAAE,KAAK,CAAC,YAAY,CAAC,GAAG,CAAC,KAC9B,KAAK,CAAC,YAAY,CAAC,GAAG,CAAC,CAAC;CAC9B;AAED,QAAA,MAAM,gBAAgB;sGAOR,MAAM,YAAY,CAAC,GAAG,CAAC;;;;CAgEpC,CAAC;AAOF,eAAe,gBAAgB,CAAC"}