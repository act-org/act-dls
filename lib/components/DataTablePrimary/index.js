"use strict";

function _typeof(obj) { "@babel/helpers - typeof"; if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var React = _interopRequireWildcard(require("react"));

var _EmptyStatePrimary = _interopRequireDefault(require("../EmptyStatePrimary"));

var _TableBase = _interopRequireDefault(require("../TableBase"));

var _TableBodyBase = _interopRequireDefault(require("../TableBodyBase"));

var _TableCellBody = _interopRequireDefault(require("../TableCellBody"));

var _TableCellHead = _interopRequireDefault(require("../TableCellHead"));

var _TableContainerPrimary = _interopRequireDefault(require("../TableContainerPrimary"));

var _TableHeadBase = _interopRequireDefault(require("../TableHeadBase"));

var _TableRowBase = _interopRequireDefault(require("../TableRowBase"));

var _styles = _interopRequireDefault(require("./styles"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _getRequireWildcardCache() { if (typeof WeakMap !== "function") return null; var cache = new WeakMap(); _getRequireWildcardCache = function _getRequireWildcardCache() { return cache; }; return cache; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

var DataTablePrimary = function DataTablePrimary(_ref) {
  var columns = _ref.columns,
      currentSortObject = _ref.currentSortObject,
      emptyStateProps = _ref.emptyStateProps,
      items = _ref.items,
      onChangeSort = _ref.onChangeSort,
      RowWrapper = _ref.RowWrapper;
  var classes = (0, _styles["default"])();
  return /*#__PURE__*/React.createElement(_TableContainerPrimary["default"], null, /*#__PURE__*/React.createElement(_TableBase["default"], null, /*#__PURE__*/React.createElement(_TableHeadBase["default"], null, /*#__PURE__*/React.createElement(_TableRowBase["default"], null, columns.map(function (column) {
    return /*#__PURE__*/React.createElement(_TableCellHead["default"], {
      currentSortObject: currentSortObject,
      key: column.label,
      onChangeSort: onChangeSort,
      sortBy: column.sortBy,
      style: column.style
    }, column.label);
  }))), /*#__PURE__*/React.createElement(_TableBodyBase["default"], null, items.map(function (item, i) {
    /* eslint-disable react/no-array-index-key */
    var children = /*#__PURE__*/React.createElement(_TableRowBase["default"], {
      hover: true,
      key: i
    }, columns.map(function (column) {
      return /*#__PURE__*/React.createElement(_TableCellBody["default"], {
        key: column.label,
        style: column.style
      }, column.renderValue(item));
    }));
    /* eslint-enable react/no-array-index-key */

    if (RowWrapper) {
      return RowWrapper(item, children);
    }

    return children;
  }))), items.length === 0 && emptyStateProps && /*#__PURE__*/React.createElement("div", {
    className: classes.emptyStateContainer
  }, /*#__PURE__*/React.createElement(_EmptyStatePrimary["default"], _extends({
    description: "No Results Found"
  }, emptyStateProps))));
}; // eslint-disable-next-line immutable/no-mutation


DataTablePrimary.defaultProps = {
  RowWrapper: undefined
};
var _default = DataTablePrimary;
exports["default"] = _default;