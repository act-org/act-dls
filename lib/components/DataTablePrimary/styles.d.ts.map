{"version":3,"file":"styles.d.ts","sourceRoot":"","sources":["../../../src/components/DataTablePrimary/styles.ts"],"names":[],"mappings":"AAAA;;GAEG;;AAMH,wBAMG"}