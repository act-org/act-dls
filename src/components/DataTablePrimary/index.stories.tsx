/**
 * @prettier
 */

/* eslint-disable sort-keys */

import * as React from 'react';
import moment from 'moment';
import { select } from '@storybook/addon-knobs';

import PackageVariant from '~/icons/PackageVariant';
import sort from '~/helpers/sort';
import SORT_DIRECTION_TYPES from '~/constants/sortDirectionTypes';
import { SortObject } from '~/types';

import DataTablePrimary from '.';

interface Item {
  id: string;
  name: string;
  updatedAt: Date;
  fieldA: string;
  fieldB: string;
  fieldC: string;
}

export const Primary = (): React.ReactElement<any> => {
  const [sortObject, setSortObject] = React.useState<SortObject>({
    sortBy: 'id',
    sortDirection: SORT_DIRECTION_TYPES.ASCENDING,
  });

  let items: Item[] = Array(
    ...Array(
      select(
        'Items Count',
        {
          0: 0,
          1: 1,
          2: 2,
          3: 3,
          4: 4,
          5: 5,
          6: 6,
          7: 7,
          8: 8,
          9: 9,
          10: 10,
        },
        5,
      ),
    ),
  ).map((_, i): any => ({
    updatedAt: moment()
      .subtract(2, 'year')
      .subtract(i + 1, 'day')
      .toDate(),
    fieldA: `Field A${i + 1}`,
    fieldB: `Field B${i + 1}`,
    fieldC: `Field C${i + 1}`,
    id: i + 1,
    name: `Item ${i + 1}`,
  }));

  // sort items
  items = items.sort(sort<Item>(sortObject));

  return (
    <DataTablePrimary<Item>
      columns={[
        {
          label: 'ID',
          renderValue: (i: Item): string => i.id,
          sortBy: 'id',
          style: {
            width: 50,
          },
        },
        {
          label: 'Name',
          renderValue: (i: Item): string => i.name,
          sortBy: 'name',
          style: {
            width: 100,
          },
        },
        {
          label: 'Field A',
          renderValue: (i: Item): string => i.fieldA,
          sortBy: 'fieldA',
          style: {
            width: 100,
          },
        },
        {
          label: 'Field B',
          renderValue: (i: Item): string => i.fieldB,
          sortBy: 'fieldB',
          style: {
            width: 100,
          },
        },
        {
          label: 'Field C',
          renderValue: (i: Item): string => i.fieldC,
          sortBy: 'fieldC',
          style: {
            width: 100,
          },
        },
        {
          label: 'Updated',
          renderValue: (i: Item): string =>
            moment(new Date(i.updatedAt)).fromNow(),
          sortBy: 'updatedAt',
          style: {
            width: 110,
          },
        },
      ]}
      currentSortObject={sortObject}
      emptyStateProps={{
        description: 'No Items Found',
        Icon: PackageVariant,
      }}
      items={items}
      onChangeSort={(newSortObject): void => {
        setSortObject(newSortObject);
      }}
    />
  );
};

export default {
  component: DataTablePrimary,
  parameters: {
    info: {
      text: 'This is the primary variant of the DataTable component.',
    },
  },
  title: 'Organisms/DataTable',
};
