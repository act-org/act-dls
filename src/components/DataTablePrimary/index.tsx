/**
 * @prettier
 */

import * as React from 'react';

import EmptyStatePrimary, {
  Props as EmptyStatePrimaryProps,
} from '~/components/EmptyStatePrimary';
import { SortObject } from '~/types';
import TableBase from '~/components/TableBase';
import TableBodyBase from '~/components/TableBodyBase';
import TableCellBody from '~/components/TableCellBody';
import TableCellHead from '~/components/TableCellHead';
import TableContainerPrimary from '~/components/TableContainerPrimary';
import TableHeadBase from '~/components/TableHeadBase';
import TableRowBase from '~/components/TableRowBase';

import useStyles from './styles';

interface Column<T> {
  renderValue: (item: T) => any;
  label: string;
  sortBy?: string;
  style?: React.CSSProperties;
}

export interface Props<T> {
  columns: Column<T>[];
  currentSortObject: SortObject;
  emptyStateProps: EmptyStatePrimaryProps;
  items: T[];
  onChangeSort: (sortObject: SortObject) => void;
  RowWrapper?: (
    item: T,
    children: React.ReactElement<any>,
  ) => React.ReactElement<any>;
}

const DataTablePrimary = <T,>({
  columns,
  currentSortObject,
  emptyStateProps,
  items,
  onChangeSort,
  RowWrapper,
}: Props<T>): React.ReactElement<any> => {
  const classes = useStyles();

  return (
    <TableContainerPrimary>
      <TableBase>
        <TableHeadBase>
          <TableRowBase>
            {columns.map(
              (column): React.ReactElement<any> => (
                <TableCellHead
                  currentSortObject={currentSortObject}
                  key={column.label}
                  onChangeSort={onChangeSort}
                  sortBy={column.sortBy}
                  style={column.style}
                >
                  {column.label}
                </TableCellHead>
              ),
            )}
          </TableRowBase>
        </TableHeadBase>

        <TableBodyBase>
          {items.map(
            (item, i): React.ReactElement<any> => {
              /* eslint-disable react/no-array-index-key */
              const children = (
                <TableRowBase hover key={i}>
                  {columns.map(
                    (column): React.ReactElement<any> => (
                      <TableCellBody key={column.label} style={column.style}>
                        {column.renderValue(item)}
                      </TableCellBody>
                    ),
                  )}
                </TableRowBase>
              );
              /* eslint-enable react/no-array-index-key */

              if (RowWrapper) {
                return RowWrapper(item, children);
              }

              return children;
            },
          )}
        </TableBodyBase>
      </TableBase>

      {items.length === 0 && emptyStateProps && (
        <div className={classes.emptyStateContainer}>
          <EmptyStatePrimary
            description="No Results Found"
            {...emptyStateProps}
          />
        </div>
      )}
    </TableContainerPrimary>
  );
};

// eslint-disable-next-line immutable/no-mutation
DataTablePrimary.defaultProps = {
  RowWrapper: undefined,
};

export default DataTablePrimary;
