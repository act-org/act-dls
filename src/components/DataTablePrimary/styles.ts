/**
 * @prettier
 */

import makeStyles from '~/styles/makeStyles';

import DIMS from '~/constants/dims';

export default makeStyles({
  emptyStateContainer: {
    padding: `${DIMS.LAYOUT_PADDING * 2.5}px ${DIMS.LAYOUT_PADDING}px
      ${DIMS.LAYOUT_PADDING}px ${DIMS.LAYOUT_PADDING}px`,
    width: '100%',
  },
});
