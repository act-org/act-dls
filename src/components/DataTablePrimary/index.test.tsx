/**
 * @prettier
 */

import * as React from 'react';
import { noop } from 'lodash';

import render from '~/helpers/test/render';

import DataTablePrimary from '.';

describe('DataTablePrimary', () => {
  it('matches the snapshot', () => {
    const { container } = render(
      <DataTablePrimary
        columns={[
          {
            dataKey: 'id',
            label: 'ID',
            searchable: true,
            sortable: true,
            style: {
              width: 75,
            },
          },
          {
            dataKey: 'name',
            label: 'Name',
            searchable: true,
            sortable: true,
            style: {
              width: 100,
            },
          },
        ]}
        items={Array(...Array(5)).map((_, i): any => ({
          id: i + 1,
          name: `Thing ${i + 1}`,
        }))}
        setSortObject={noop}
        sortObject={{
          dataKey: 'id',
          direction: 'ASC',
        }}
      />,
    );

    expect(container).toMatchSnapshot();
  });
});
