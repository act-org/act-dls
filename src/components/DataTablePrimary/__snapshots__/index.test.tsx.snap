// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataTablePrimary matches the snapshot 1`] = `
<div>
  <div
    class="MuiPaper-root makeStyles-paperRoot-2 MuiTableContainer-root MuiPaper-elevation0 MuiPaper-rounded"
  >
    <table
      class="MuiTable-root"
    >
      <thead
        class="MuiTableHead-root"
      >
        <tr
          class="MuiTableRow-root MuiTableRow-head"
        >
          <div
            class="MuiTableCell-root makeStyles-tableCellRoot-5 makeStyles-tableCellRootSortable-6 makeStyles-tableCellRootSortApplied-7 MuiTableCell-head"
            role="columnheader"
            scope="col"
            style="width: 50px;"
          >
            <p
              class="MuiTypography-root makeStyles-typographyRoot-8 MuiTypography-body1"
            >
              ID
            </p>
            <div
              class="MuiGrid-root makeStyles-sortContainerRoot-3 MuiGrid-container"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root makeStyles-sortIconRoot-4"
                focusable="false"
                role="button"
                style="color: rgb(255, 255, 255);"
                viewBox="9 3 5 15"
              >
                <path
                  d="M7.41,15.41L12,10.83L16.59,15.41L18,14L12,8L6,14L7.41,15.41Z"
                />
              </svg>
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root makeStyles-sortIconRoot-4"
                focusable="false"
                role="button"
                viewBox="9 7 5 15"
              >
                <path
                  d="M7.41,8.58L12,13.17L16.59,8.58L18,10L12,16L6,10L7.41,8.58Z"
                />
              </svg>
            </div>
          </div>
          <div
            class="MuiTableCell-root makeStyles-tableCellRoot-5 makeStyles-tableCellRootSortable-6 MuiTableCell-head"
            role="columnheader"
            scope="col"
            style="width: 100px;"
          >
            <p
              class="MuiTypography-root makeStyles-typographyRoot-8 MuiTypography-body1"
            >
              Name
            </p>
            <div
              class="MuiGrid-root makeStyles-sortContainerRoot-3 MuiGrid-container"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root makeStyles-sortIconRoot-4"
                focusable="false"
                role="button"
                viewBox="9 3 5 15"
              >
                <path
                  d="M7.41,15.41L12,10.83L16.59,15.41L18,14L12,8L6,14L7.41,15.41Z"
                />
              </svg>
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root makeStyles-sortIconRoot-4"
                focusable="false"
                role="button"
                viewBox="9 7 5 15"
              >
                <path
                  d="M7.41,8.58L12,13.17L16.59,8.58L18,10L12,16L6,10L7.41,8.58Z"
                />
              </svg>
            </div>
          </div>
          <div
            class="MuiTableCell-root makeStyles-tableCellRoot-5 makeStyles-tableCellRootSortable-6 MuiTableCell-head"
            role="columnheader"
            scope="col"
            style="width: 100px;"
          >
            <p
              class="MuiTypography-root makeStyles-typographyRoot-8 MuiTypography-body1"
            >
              Field A
            </p>
            <div
              class="MuiGrid-root makeStyles-sortContainerRoot-3 MuiGrid-container"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root makeStyles-sortIconRoot-4"
                focusable="false"
                role="button"
                viewBox="9 3 5 15"
              >
                <path
                  d="M7.41,15.41L12,10.83L16.59,15.41L18,14L12,8L6,14L7.41,15.41Z"
                />
              </svg>
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root makeStyles-sortIconRoot-4"
                focusable="false"
                role="button"
                viewBox="9 7 5 15"
              >
                <path
                  d="M7.41,8.58L12,13.17L16.59,8.58L18,10L12,16L6,10L7.41,8.58Z"
                />
              </svg>
            </div>
          </div>
          <div
            class="MuiTableCell-root makeStyles-tableCellRoot-5 makeStyles-tableCellRootSortable-6 MuiTableCell-head"
            role="columnheader"
            scope="col"
            style="width: 100px;"
          >
            <p
              class="MuiTypography-root makeStyles-typographyRoot-8 MuiTypography-body1"
            >
              Field B
            </p>
            <div
              class="MuiGrid-root makeStyles-sortContainerRoot-3 MuiGrid-container"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root makeStyles-sortIconRoot-4"
                focusable="false"
                role="button"
                viewBox="9 3 5 15"
              >
                <path
                  d="M7.41,15.41L12,10.83L16.59,15.41L18,14L12,8L6,14L7.41,15.41Z"
                />
              </svg>
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root makeStyles-sortIconRoot-4"
                focusable="false"
                role="button"
                viewBox="9 7 5 15"
              >
                <path
                  d="M7.41,8.58L12,13.17L16.59,8.58L18,10L12,16L6,10L7.41,8.58Z"
                />
              </svg>
            </div>
          </div>
          <div
            class="MuiTableCell-root makeStyles-tableCellRoot-5 makeStyles-tableCellRootSortable-6 MuiTableCell-head"
            role="columnheader"
            scope="col"
            style="width: 100px;"
          >
            <p
              class="MuiTypography-root makeStyles-typographyRoot-8 MuiTypography-body1"
            >
              Field C
            </p>
            <div
              class="MuiGrid-root makeStyles-sortContainerRoot-3 MuiGrid-container"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root makeStyles-sortIconRoot-4"
                focusable="false"
                role="button"
                viewBox="9 3 5 15"
              >
                <path
                  d="M7.41,15.41L12,10.83L16.59,15.41L18,14L12,8L6,14L7.41,15.41Z"
                />
              </svg>
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root makeStyles-sortIconRoot-4"
                focusable="false"
                role="button"
                viewBox="9 7 5 15"
              >
                <path
                  d="M7.41,8.58L12,13.17L16.59,8.58L18,10L12,16L6,10L7.41,8.58Z"
                />
              </svg>
            </div>
          </div>
        </tr>
      </thead>
      <tbody
        class="MuiTableBody-root"
      >
        <tr
          class="MuiTableRow-root MuiTableRow-hover"
        >
          <td
            class="MuiTableCell-root makeStyles-tableCellRoot-9 MuiTableCell-body MuiTableCell-paddingNone"
            role="cell"
            scope="row"
            style="width: 50px;"
          >
            1
          </td>
          <td
            class="MuiTableCell-root makeStyles-tableCellRoot-9 MuiTableCell-body MuiTableCell-paddingNone"
            role="cell"
            scope="row"
            style="width: 100px;"
          >
            Item 1
          </td>
          <td
            class="MuiTableCell-root makeStyles-tableCellRoot-9 MuiTableCell-body MuiTableCell-paddingNone"
            role="cell"
            scope="row"
            style="width: 100px;"
          >
            Field A1
          </td>
          <td
            class="MuiTableCell-root makeStyles-tableCellRoot-9 MuiTableCell-body MuiTableCell-paddingNone"
            role="cell"
            scope="row"
            style="width: 100px;"
          >
            Field B1
          </td>
          <td
            class="MuiTableCell-root makeStyles-tableCellRoot-9 MuiTableCell-body MuiTableCell-paddingNone"
            role="cell"
            scope="row"
            style="width: 100px;"
          >
            Field C1
          </td>
        </tr>
        <tr
          class="MuiTableRow-root MuiTableRow-hover"
        >
          <td
            class="MuiTableCell-root makeStyles-tableCellRoot-9 MuiTableCell-body MuiTableCell-paddingNone"
            role="cell"
            scope="row"
            style="width: 50px;"
          >
            2
          </td>
          <td
            class="MuiTableCell-root makeStyles-tableCellRoot-9 MuiTableCell-body MuiTableCell-paddingNone"
            role="cell"
            scope="row"
            style="width: 100px;"
          >
            Item 2
          </td>
          <td
            class="MuiTableCell-root makeStyles-tableCellRoot-9 MuiTableCell-body MuiTableCell-paddingNone"
            role="cell"
            scope="row"
            style="width: 100px;"
          >
            Field A2
          </td>
          <td
            class="MuiTableCell-root makeStyles-tableCellRoot-9 MuiTableCell-body MuiTableCell-paddingNone"
            role="cell"
            scope="row"
            style="width: 100px;"
          >
            Field B2
          </td>
          <td
            class="MuiTableCell-root makeStyles-tableCellRoot-9 MuiTableCell-body MuiTableCell-paddingNone"
            role="cell"
            scope="row"
            style="width: 100px;"
          >
            Field C2
          </td>
        </tr>
        <tr
          class="MuiTableRow-root MuiTableRow-hover"
        >
          <td
            class="MuiTableCell-root makeStyles-tableCellRoot-9 MuiTableCell-body MuiTableCell-paddingNone"
            role="cell"
            scope="row"
            style="width: 50px;"
          >
            3
          </td>
          <td
            class="MuiTableCell-root makeStyles-tableCellRoot-9 MuiTableCell-body MuiTableCell-paddingNone"
            role="cell"
            scope="row"
            style="width: 100px;"
          >
            Item 3
          </td>
          <td
            class="MuiTableCell-root makeStyles-tableCellRoot-9 MuiTableCell-body MuiTableCell-paddingNone"
            role="cell"
            scope="row"
            style="width: 100px;"
          >
            Field A3
          </td>
          <td
            class="MuiTableCell-root makeStyles-tableCellRoot-9 MuiTableCell-body MuiTableCell-paddingNone"
            role="cell"
            scope="row"
            style="width: 100px;"
          >
            Field B3
          </td>
          <td
            class="MuiTableCell-root makeStyles-tableCellRoot-9 MuiTableCell-body MuiTableCell-paddingNone"
            role="cell"
            scope="row"
            style="width: 100px;"
          >
            Field C3
          </td>
        </tr>
        <tr
          class="MuiTableRow-root MuiTableRow-hover"
        >
          <td
            class="MuiTableCell-root makeStyles-tableCellRoot-9 MuiTableCell-body MuiTableCell-paddingNone"
            role="cell"
            scope="row"
            style="width: 50px;"
          >
            4
          </td>
          <td
            class="MuiTableCell-root makeStyles-tableCellRoot-9 MuiTableCell-body MuiTableCell-paddingNone"
            role="cell"
            scope="row"
            style="width: 100px;"
          >
            Item 4
          </td>
          <td
            class="MuiTableCell-root makeStyles-tableCellRoot-9 MuiTableCell-body MuiTableCell-paddingNone"
            role="cell"
            scope="row"
            style="width: 100px;"
          >
            Field A4
          </td>
          <td
            class="MuiTableCell-root makeStyles-tableCellRoot-9 MuiTableCell-body MuiTableCell-paddingNone"
            role="cell"
            scope="row"
            style="width: 100px;"
          >
            Field B4
          </td>
          <td
            class="MuiTableCell-root makeStyles-tableCellRoot-9 MuiTableCell-body MuiTableCell-paddingNone"
            role="cell"
            scope="row"
            style="width: 100px;"
          >
            Field C4
          </td>
        </tr>
        <tr
          class="MuiTableRow-root MuiTableRow-hover"
        >
          <td
            class="MuiTableCell-root makeStyles-tableCellRoot-9 MuiTableCell-body MuiTableCell-paddingNone"
            role="cell"
            scope="row"
            style="width: 50px;"
          >
            5
          </td>
          <td
            class="MuiTableCell-root makeStyles-tableCellRoot-9 MuiTableCell-body MuiTableCell-paddingNone"
            role="cell"
            scope="row"
            style="width: 100px;"
          >
            Item 5
          </td>
          <td
            class="MuiTableCell-root makeStyles-tableCellRoot-9 MuiTableCell-body MuiTableCell-paddingNone"
            role="cell"
            scope="row"
            style="width: 100px;"
          >
            Field A5
          </td>
          <td
            class="MuiTableCell-root makeStyles-tableCellRoot-9 MuiTableCell-body MuiTableCell-paddingNone"
            role="cell"
            scope="row"
            style="width: 100px;"
          >
            Field B5
          </td>
          <td
            class="MuiTableCell-root makeStyles-tableCellRoot-9 MuiTableCell-body MuiTableCell-paddingNone"
            role="cell"
            scope="row"
            style="width: 100px;"
          >
            Field C5
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
